### PR TITLE
Fix saliency maps order for h-cls in cpp

### PR DESCRIPTION
**Pull request by sovrasov**
# What does this PR do?

The order could have been twisted, because logits in hierarchical classification are not placed in the same order as "labels" in model info.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: model_api/cpp/models/src/classification_model.cpp [hunk 1]
```diff
@@ -358,6 +359,27 @@ std::unique_ptr<ResultBase> ClassificationModel::get_hierarchical_predictions(In
     return retVal;
 }
 
+ov::Tensor ClassificationModel::reorder_saliency_maps(const ov::Tensor& source_maps) {
+    if (!hierarchical || source_maps.get_shape().size() == 1) {
```
Comment (GalyaZalesskaya) [inline model_api/cpp/models/src/classification_model.cpp:363]: In which case source_maps.get_shape().size()==1 can be  true?

File: model_api/cpp/models/src/classification_model.cpp [hunk 2]
```diff
@@ -358,6 +359,27 @@ std::unique_ptr<ResultBase> ClassificationModel::get_hierarchical_predictions(In
...
+
+    auto reordered_maps = ov::Tensor(source_maps.get_element_type(), source_maps.get_shape());
+    const std::uint8_t* source_maps_ptr = static_cast<std::uint8_t*>(source_maps.data());
+    std::uint8_t* reordered_maps_ptr = static_cast<std::uint8_t*>(reordered_maps.data());
+
+    size_t shape_offset = (source_maps.get_shape().size() == 4) ? 1 : 0;
+    size_t map_byte_size = source_maps.get_element_type().size() *
+        source_maps.get_shape()[shape_offset + 1] * source_maps.get_shape()[shape_offset + 2];
+
+    for (size_t i = 0; i < source_maps.get_shape()[shape_offset]; ++i) {
```
Comment (GalyaZalesskaya) [inline model_api/cpp/models/src/classification_model.cpp:375]: Does it work if there are several maps returned, so source_maps have batch size 2? For instance (2,3,7,7) for 2 images and 3 classes

File: model_api/cpp/models/src/classification_model.cpp (hunk 1 above)
Comment (sovrasov) [reply model_api/cpp/models/src/classification_model.cpp:363]: If saliency map is an empty tensor, that's a sanity check

**Review by GalyaZalesskaya — APPROVED** [review]

**Comment by sovrasov** [comment]
cpp part is finem so we can merge this
